test/test_typing.py

```diff
@@ -0,0 +1,150 @@
+# based on NumPy numpy/typing/tests/test_typing.py
+
+import itertools
+import os
+import re
+import shutil
+from typing import IO, Dict, List
+
+import pytest
+import numpy as np
+
+try:
+    from mypy import api
+except ImportError:
+    NO_MYPY = True
+else:
+    NO_MYPY = False
+
+
+DATA_DIR = os.path.join(os.path.dirname(__file__), "typing")
+REVEAL_DIR = os.path.join(DATA_DIR, "reveal")
+MYPY_INI = os.path.join(DATA_DIR, os.pardir, os.pardir, "mypy.ini")
+CACHE_DIR = os.path.join(DATA_DIR, ".mypy_cache")
+
+#: A dictionary with file names as keys and lists of the mypy stdout as values.
+#: To-be populated by `run_mypy`.
+OUTPUT_MYPY: Dict[str, List[str]] = {}
+
+
+def _key_func(key: str) -> str:
+    """Split at the first occurance of the ``:`` character.
+
+    Windows drive-letters (*e.g.* ``C:``) are ignored herein.
+    """
+    drive, tail = os.path.splitdrive(key)
+    return os.path.join(drive, tail.split(":", 1)[0])
+
+
+@pytest.mark.skipif(NO_MYPY, reason="Mypy is not installed")
+@pytest.fixture(scope="module", autouse=True)
+def run_mypy() -> None:
+    """Clears the cache and run mypy before running any of the typing tests.
+
+    The mypy results are cached in `OUTPUT_MYPY` for further use.
+
+    """
+    if os.path.isdir(CACHE_DIR):
+        shutil.rmtree(CACHE_DIR)
```

Comment:
Can you add a comment why cache needs to be invalidated before running the tests?
MyPy should be capable of updating the cache if files have changed between the runs, shouldn't it?

Reply:
This was also inherited from NumPy source code. But from my personal experience, 99% of the time mypy is capable of invalidating the cache. Only in a few specific scenarios this wasn't true.